### PR TITLE
M870

### PR DESCRIPTION
**Pull request by gwan93**
This feature implemented a persistent button to show how many projects passed the filters. There is also an 'x' on the button to clear all filters.

The logic to clear filters already existed in the FilterPane component, but in order to give this functionality to the clear all filters button, the logic had to be elevated into a useContext

The order of the clear filter button had to come before the ViewToggle and Zoom to Selected Data/Filtered Data buttons, so a new component called MapAndTableControls was created to contain all these buttons

### CODE REVIEW

**Comment by gwan93** [comment]
[Persistient "filters applied" notification and ability to clear all.](https://trello.com/c/QeCcsHaJ/870-persistient-filters-applied-notification-and-ability-to-clear-all)

**Review by alanjleonard — CHANGES_REQUESTED** [review]
Looks good, however, it should also apply to the project filters as well. So if I type `fiji` in the project filter, the number of projects filtered should be reflected in the persistent notification. Clearing the notification should also clear that input.

File: src/components/MapAndTableControls.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    map.fitBounds(bounds)
+  }
+
+  const handleZoomToSelectedSite = () => {
+    if (!map) {
+      return
+    }
+
+    const queryParams = new URLSearchParams(location.search)
+    if (queryParams.has('sample_event_id')) {
```
Comment (julia-rom) [inline src/components/MapAndTableControls.jsx:65]: can this be in a named function?

File: src/components/MapAndTableControls.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  gap: 0.5rem;
+  align-items: center;
+`
+
+export default function MapAndTableControls({ map = undefined, view, setView }) {
+  const { displayedProjects, projectDataCount } = useFilterProjectsContext()
+  const { isDesktopWidth } = useResponsive()
+
+  const handleZoomToFilteredData = () => {
+    if (!map) {
```
Comment (julia-rom) [inline src/components/MapAndTableControls.jsx:42]: would suggest 

```
if (!map || !displayedProjects || displayedProjects.length === 0) {
      return
    }
```

File: src/components/MermaidDash.jsx [hunk 3]
```diff
@@ -125,77 +125,66 @@ const MobileFooterContainer = styled('div')`
...
   const getAuthorizationHeaders = async (getAccessTokenSilently) => ({
     headers: {
       Authorization: `Bearer ${await getAccessTokenSilently()}`,
     },
   })
 
+  const fetchData = useCallback(
+    async (token = '') => {
+      try {
+        let nextPageUrl = `${import.meta.env.VITE_REACT_APP_MERMAID_API_ENDPOINT}?limit=300&page=1`
```
Comment (julia-rom) [inline src/components/MermaidDash.jsx:149]: would separate this out a bit for greater readability:

```
const apiEndpoint = import.meta.env.VITE_REACT_APP_MERMAID_API_ENDPOINT
    const initialUrl = `${apiEndpoint}?limit=300&page=1`
```

Comment (julia-rom) [reply src/components/MermaidDash.jsx:149]: `let nextPageUrl = initialUrl`

File: src/components/MermaidDash.jsx [hunk 4]
```diff
@@ -125,77 +125,66 @@ const MobileFooterContainer = styled('div')`
...
+
+        while (nextPageUrl !== null) {
```
Comment (julia-rom) [inline src/components/MermaidDash.jsx:151]: could simplify the null check .. 

```
while (nextPageUrl) { 
        const response = await fetch(nextPageUrl, {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        });

        if (!response.ok) {
          console.error('Failed to fetch data:', response.status, response.statusText)
          break
        }

        const data = await response.json()
``` ...

**Review by julia-rom — CHANGES_REQUESTED** [review]

File: src/components/MapAndTableControls.jsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      'method',
+      'dataSharing',
+      'projects',
+    ]
+
+    return filterKeys.some((key) => queryParams.has(key))
+  }
+
+  const hasSelectedSite = () => {
+    const queryParams = new URLSearchParams(location.search)
```
[the remark this answers is not in the document]
Comment (julia-rom) [reply src/components/MapAndTableControls.jsx:94]: and place `const queryParams = new URLSearchParams(location.search)` under ```
const { displayedProjects, projectDataCount } = useFilterProjectsContext()
  const { isDesktopWidth } = useResponsive()
```

**Review by julia-rom — APPROVED** [review]

**Pull request merged**